pkg/vault/test/foundry/CompositeLiquidityRouterNestedPools.t.sol

```diff
@@ -0,0 +1,358 @@
+import { BaseVaultTest } from "./utils/BaseVaultTest.sol";
+
+contract CompositeLiquidityRouterNestedPools is BaseVaultTest {
+    using ArrayHelpers for *;
+    using CastingHelpers for address[];
+    using FixedPoint for uint256;
+
+    address internal parentPool;
+    address internal childPoolA;
+    address internal childPoolB;
+
+    // Max of 5 wei of error when retrieving tokens from a nested pool.
+    uint256 internal constant MAX_ROUND_ERROR = 5;
+
+    function setUp() public override {
+        BaseVaultTest.setUp();
+
+        childPoolA = _createPool([address(usdc), address(weth)].toMemoryArray(), "childPoolA");
+        childPoolB = _createPool([address(wsteth), address(dai)].toMemoryArray(), "childPoolB");
+        parentPool = _createPool(
+            [address(childPoolA), address(childPoolB), address(dai)].toMemoryArray(),
+            "parentPool"
+        );
+
+        vm.startPrank(lp);
+        uint256 childPoolABptOut = _initPool(childPoolA, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        uint256 childPoolBBptOut = _initPool(childPoolB, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+
+        uint256[] memory tokenIndexes = getSortedIndexes([childPoolA, childPoolB, address(dai)].toMemoryArray());
+        uint256 poolAIdx = tokenIndexes[0];
+        uint256 poolBIdx = tokenIndexes[1];
+        uint256 daiIdx = tokenIndexes[2];
+
+        uint256[] memory amountsInParentPool = new uint256[](3);
+        amountsInParentPool[daiIdx] = poolInitAmount;
+        amountsInParentPool[poolAIdx] = childPoolABptOut;
+        amountsInParentPool[poolBIdx] = childPoolBBptOut;
+        vm.stopPrank();
+
+        approveForPool(IERC20(childPoolA));
+        approveForPool(IERC20(childPoolB));
+        approveForPool(IERC20(parentPool));
+
+        vm.startPrank(lp);
+        _initPool(parentPool, amountsInParentPool, 0);
+        vm.stopPrank();
+    }
+
+    function testRemoveLiquidityNestedPool__Fuzz(uint256 proportionToRemove) public {
+        // Remove between 0.0001% and 50% of each pool liquidity.
+        proportionToRemove = bound(proportionToRemove, 1e12, 50e16);
+
+        uint256 totalPoolBPTs = BalancerPoolToken(parentPool).totalSupply();
+        // Since LP is the owner of all BPT supply, and part of the BPTs were burned in the initialization step, using
+        // totalSupply is more accurate to remove exactly the proportion that we intend from each pool.
+        uint256 exactBptIn = totalPoolBPTs.mulDown(proportionToRemove);
+
+        NestedPoolTestLocals memory vars = _createNestedPoolTestLocals();
+
+        // During pool initialization, POOL_MINIMUM_TOTAL_SUPPLY amount of BPT is burned to address(0), so that the
+        // pool cannot be completely drained. We need to discount this amount of tokens from the total liquidity that
+        // we can extract from the child pools.
+        uint256 deadTokens = (POOL_MINIMUM_TOTAL_SUPPLY / 2).mulDown(proportionToRemove);
```

Comment:
hm. This is correct, but if you don't want to deal with this another option is having `alice` add liquidity after initialization. Then, alice can remove all the added tokens, without any dead portion.

Reply:
I tried to replace `lp` by `alice`, and I still needed to deal with deadTokens because "removeLiquidityProportional" takes into account Alice's BPTs and `totalSupply`, which includes burned tokens.